lib/improver/tests/weights/test_WeightsUtilities.py

```diff
@@ -0,0 +1,394 @@
+                                                       missing_weights)
+        expected_result = np.array([0.44978518, 0.28195489,
+                                    -1.0, 0.12083781,
+                                    0.08458647, 0.06283566])
+        self.assertArrayAlmostEqual(result, expected_result)
+
+    def test_returns_correct_values_proportional(self):
+        """Test it returns the correct values, method is proportional."""
+        weights_in = np.array([0.41957573, 0.25174544,
+                               0.15104726, 0.09062836,
+                               0.05437701, 0.03262621])
+        missing_weights = np.ones(6)
+        missing_weights[2] = 0.0
+        result = WeightsUtilities.redistribute_weights(weights_in,
+                                                       missing_weights,
+                                                       method='proportional')
+        expected_result = np.array([0.49422742, 0.29653645,
+                                    -1.0, 0.10675312,
+                                    0.06405187, 0.03843112])
+        self.assertArrayAlmostEqual(result, expected_result)
+
+
+class Test_nonlinear_weights(IrisTest):
+    """Test the nonlinear weights function. """
+
+    def test_basic(self):
+        """Test that the function returns an array of weights. """
+        result = WeightsUtilities.nonlinear_weights(3, 0.85)
+        self.assertIsInstance(result, np.ndarray)
+
+    def test_fails_num_of_weights_set_wrong(self):
+        """Test it fails if num_of_weights not an integer or > 0. """
+        msg = ('Number of weights must be integer > 0')
+        with self.assertRaisesRegexp(ValueError, msg):
+            WeightsUtilities.nonlinear_weights(3.0, 0.85)
+        with self.assertRaisesRegexp(ValueError, msg):
+            WeightsUtilities.nonlinear_weights(-1, 0.85)
+
+    def test_fails_cval_set_wrong(self):
+        """Test it fails if < """
+        msg = ('cval must be greater than 0.0')
+        with self.assertRaisesRegexp(ValueError, msg):
+            WeightsUtilities.nonlinear_weights(3, -0.1)
+        with self.assertRaisesRegexp(ValueError, msg):
+            WeightsUtilities.nonlinear_weights(3, 1.85)
+
+    def test_returns_correct_values(self):
+        """Test it returns the correct values for num_of_weights 6, cval 0.6"""
+        result = WeightsUtilities.nonlinear_weights(6, 0.6)
+        expected_result = np.array([0.41957573, 0.25174544,
+                                    0.15104726, 0.09062836,
+                                    0.05437701, 0.03262621])
+        self.assertArrayAlmostEqual(result, expected_result)
+
+
+class Test_linear_weights(IrisTest):
+    """Test the linear weights function. """
+
+    def test_basic(self):
+        """Test that the function returns an array of weights. """
+        result = WeightsUtilities.linear_weights(3)
+        self.assertIsInstance(result, np.ndarray)
+
+    def test_fails_num_of_weights_set_wrong(self):
```

Comment:
As this check hasmoved to `process_coord` there is no way for the error to arise. The value will always be greater than 0 because a coordinate.points array can not have negative length.